hack/generator/azure-arm.yaml

```diff
@@ -203,6 +203,12 @@ typeTransformers:
   # only readonly properties that have all been removed in the swagger
   # -> deployment template conversion.
 
+  - group: deploymenttemplate
+    name: ResourceLocations
+    target:
+      name: string
+    because: Modeling this as an enum doesn't work well in the context of CRDs because new regions are regularly added
```

Comment:
can we rewrite this to add (instead of replace) `string`? it's nice to have the current list for autocomplete

probably we need to do some work to enable this

Reply:
Not sure I follow what you're saying here? You mean to add an element to `oneOf` so it's always `oneOf` the `enum` and `string`?

That's actually what it is on ResourceBase I think - the problem is the resulting type is pretty awkward to work with.